modules/action-gateway/src/main.cpp

```diff
@@ -754,6 +762,127 @@ class Gateway : public RFModule
         return false;
     }
 
+    /****************************************************************/
+    bool pouringApproach(const Vector &target, const Vector &neck_position,
+                         const string &part="select")
+    {
+        if ((target.length()<3) || (neck_position.length()!=3))
+        {
+            yError()<<"Invalid parameters given for pouring approach";
+            return false;
+        }
+
+        std:string part_=choosePart(part, target);
```

Comment:
this should be `std::string`, although I would rather use `string` plainly to be consistent with the rest of the code here.